File: algorithm/insertion_sort/insertion_sort.cpp
#include <cstdio>
using namespace std;

int* insertion_sort(int *arr, int size);

int main(void) {

	int size;
	scanf("%d", &size);

	int* arr = new int[size];

	for (int i = 0; i < size; i++)
		scanf("%d", &arr[i]);

	for (int i = 0; i < size; i++)
		printf("[%d]", arr[i]);
	printf("\n");

	arr = insertion_sort(arr, size);

	for (int i = 0; i < size; i++)
		printf("[%d]", arr[i]);
}

int* insertion_sort(int *arr, int size) {

	for (int i = 0; i < size; i++) {
		int value = arr[i];
		int index = i;

		while (index > 0 && value < arr[index - 1]) {
			arr[index] = arr[index - 1];
			index--;
		}
		arr[index] = value;
	}

	return arr;
}

File: algorithm/insertion_sort/InsertionSort.java
package insertion_sort;

import java.util.Scanner;

public class InsertionSort {

	public static void main(String[] args) {

		Scanner sc = new Scanner(System.in);

		int size = sc.nextInt();
		int[] arr = new int[size];

		for (int i = 0; i < size; i++) {
			arr[i] = sc.nextInt();
		}

		for (int n : arr)
			System.out.print("[" + n + "]");
		System.out.println();

		arr = insertionSort(arr);

		for (int n : arr)
			System.out.print("[" + n + "]");
		System.out.println();
	}

	public static int[] insertionSort(int[] arr) {

		for (int i = 1; i < arr.length; i++) {
			int value = arr[i];
			int index = i;
			
			while(index > 0 && value < arr[index-1]) {
				arr[index] = arr[index-1];
				index--;
			}
			arr[index] = value;
		}

		return arr;
	}
}
